src/client/sandbox/shadow-ui.js

```diff
@@ -405,29 +438,21 @@ export default class ShadowUI extends SandboxBase {
         }
     }
 
-    static isShadowContainer (el) {
-        if (domUtils.isDomElement(el))
-            return domUtils.isBodyElement(el) || domUtils.isHeadElement(el);
-
-        return false;
-    }
-
-    static isShadowContainerCollection (collection) {
+    static _checkFlag (obj, flag) {
         try {
-            if (collection && collection.length && !domUtils.isWindow(collection) && collection[0] &&
-                collection[0].nodeType) {
-                var parent = collection[0].parentNode;
-
-                if (parent && (parent.childNodes === collection || parent.children === collection))
-                    return ShadowUI.isShadowContainer(parent);
-            }
+            return obj[flag];
         }
-            /*eslint-disable no-empty */
         catch (e) {
+            return false;
         }
-        /*eslint-enable no-empty */
+    }
 
-        return false;
+    static isShadowContainer (el) {
+        return ShadowUI._checkFlag(el, IS_SHADOW_CONTAINER_FLAG);
```

Comment:
This PR is a performace optimization.
Also, it would be good to measure 'Bottom-Up' time (see Google Chrome Dev Tools, panel - Performance) for web page described in issue